iamy.go

```diff
@@ -85,3 +94,29 @@ func init() {
 	}
 	defaultDir = filepath.Clean(dir)
 }
+
+func performVersionChecks() {
+	currentIAMyVersion, _ := semver.Make(Version)
+	log.Printf("current version is %s", currentIAMyVersion)
+
+	if _, err := os.Stat(versionFileName); !os.IsNotExist(err) {
+		log.Printf("%s found", versionFileName)
+		fileBytes, _ := ioutil.ReadFile(versionFileName)
+		fileContents := string(fileBytes)
+
+		if fileContents != "" {
+			re := regexp.MustCompile(`\d\.\d+\.\d`)
+			match := re.FindStringSubmatch(fileContents)
+			localDesiredVersion, _ := semver.Make(match[0])
+			log.Printf("local project wants version %s", localDesiredVersion)
+
+			// We don't want to notify users if the `Version` is "dev" as it's not
+			// actually too old. It could be that they are running non-released
+			// versions.
+			if Version != "dev" && currentIAMyVersion.LE(localDesiredVersion) {
+				fmt.Printf(versionTooOldError, currentIAMyVersion, localDesiredVersion)
```

Comment:
should this be on stderr?

Reply:
It can be if you have an intended use for it. I didn't so `stdout` it was. It will exit with a status code of 1 if that's any help here.